src/components/ContentNode/Caption.vue

```diff
@@ -0,0 +1,87 @@
+<!--
+  This source file is part of the Swift.org open source project
+
+  Copyright (c) 2021-2023 Apple Inc. and the Swift project authors
+  Licensed under Apache License v2.0 with Runtime Library Exception
+
+  See https://swift.org/LICENSE.txt for license information
+  See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+-->
+
+<template>
+  <component class="caption" :class="{ centered }" :is="tag">
+    <template v-if="title">
+      <strong>{{ title }}</strong>&nbsp;<slot />
+    </template>
+    <template v-else>
+      <slot />
+    </template>
+  </component>
+</template>
+
+<script>
+const CaptionTag = {
+  caption: 'caption',
+  figcaption: 'figcaption',
+};
+
+export default {
+  name: 'Caption',
+  constants: { CaptionTag },
+  props: {
+    title: {
+      type: String,
+      required: false,
+    },
+    centered: {
+      type: Boolean,
+      default: false,
+    },
+    tag: {
+      type: String,
+      required: true,
+      validator: v => Object.hasOwnProperty.call(CaptionTag, v),
+    },
+  },
+};
+</script>
+
+<style scoped lang="scss">
+@import 'docc-render/styles/_core.scss';
+
+.caption {
+  @include font-styles(documentation-caption);
+
+  &.centered {
+    text-align: center;
+  }
+}
+
+// `space-out-between-siblings` helper would normally be used, but it won't work
+// for certain things like <picture> or <thead>/<tbody> with different
+// display types, so the spacing is manually done here for both kinds of
+// captions
+figcaption {
+  &:first-child {
+    margin: 0 0 var(--spacing-stacked-margin-large) 0;
+  }
+  &:last-child {
+    margin: var(--spacing-stacked-margin-large) 0 0 0;
+  }
+}
+caption {
+  margin: 0 0 var(--spacing-stacked-margin-large) 0;
+
+  // `caption-side` must be used for the table version of <caption> to appear
+  // underneath the table since the element must always be the first element
+  // within the <table> in the DOM according to the spec
+  &.centered {
+    caption-side: bottom;
+    margin: var(--spacing-stacked-margin-large) 0 0 0;
+  }
+}
```

Comment:
I believe we could simplify all these styles into:

```scss
.caption {
  @include font-styles(documentation-caption);
  margin: 0 0 var(--spacing-stacked-margin-large) 0;

  &.centered {
    text-align: center;
    margin: var(--spacing-stacked-margin-large) 0 0 0;
    // caption-side only applies for tables
    caption-side: bottom;
  }
}
```
because in both `figcaption` and `caption` cases, we have 2 styles possible: on top-left (default) or bottom-centered (when we add the .centered class).

if we are worried that an extra margin would be add in case there is no siblings, we could add:
```scss
&:only-child {
  margin: 0;
}
```
but I don't think this use case will ever happen.

Reply:
Hm, I'll see if I can simplify it more, but I'm not sure we would be able to use that suggestion as-is. The `caption-side` is supposed to only apply to `<caption>` elements and not `<figcaption>` ones.

Also, it would be nice to have the margin rules dependent on siblings just to make it a little more flexible if someone wanted to use the component elsewhere, but maybe that's not a big deal since it should always be used with siblings anyway.

I'll see if it makes sense to simplify more though.

Reply:
I think this is the best I can come up with at the moment: https://github.com/apple/swift-docc-render/pull/706/commits/54a6137a9bb80c3789cac7bb35cf52caa40c498f

Reply:
Looks good!